geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/records/SimpleFeatureRecordConverter.scala

```diff
@@ -34,17 +36,93 @@ object SimpleFeatureRecordConverter {
 
   import scala.collection.JavaConverters._
 
-  def apply(sft: SimpleFeatureType, encoding: GeometryEncoding = GeometryEncoding.Wkt): SimpleFeatureRecordConverter = {
+  def fromSFT(sft: SimpleFeatureType, encoding: GeometryEncoding = GeometryEncoding.Wkt): SimpleFeatureRecordConverter = {
     val converters = sft.getAttributeDescriptors.asScala.map { descriptor =>
       getConverter(descriptor.getLocalName, ObjectType.selectType(descriptor), encoding)
+    }.toArray
+
+    val schema: RecordSchema = getRecordSchema(sft, converters)
+    new SimpleFeatureRecordConverterImpl(sft, schema, converters)
+  }
+
+  def fromRecordSchema(schema: RecordSchema,
+                       encodings: scala.collection.Map[String, TypeAndEncoding],
+                       encoding: GeometryEncoding = GeometryEncoding.Wkt,
+                       defaultGeometryColumn: Option[String] = None,
+                       typeName: Option[String] = None): SimpleFeatureRecordConverter = {
+    val sft: SimpleFeatureType = recordSchemaToSFT(schema, encodings, defaultGeometryColumn, typeName)
+    val converters = sft.getAttributeDescriptors.asScala.map { descriptor =>
+      getConverter(descriptor.getLocalName, ObjectType.selectType(descriptor), encoding, encodings)
+    }.toArray
+
+    new SimpleFeatureRecordConverterImpl(sft, schema, converters)
+  }
+
+  private def recordSchemaToSFT(schema: RecordSchema,
+                                encodings: scala.collection.Map[String, TypeAndEncoding],
+                                defaultGeometryColumn: Option[String] = None,
+                                typeName: Option[String] = None): SimpleFeatureType = {
+    schema match {
+      case sftSchema: SimpleFeatureTypeRecordSchema => sftSchema.sft
+      case _ => 
+        // If we do not have enough information, throw an exception
+        if (encodings.isEmpty) {
+          throw new Exception(s"No geometry columns were configured.")
+        }
+        val builder = new SimpleFeatureTypeBuilder
+
+        schema.getFields.asScala.foreach { field =>
+          val name = field.getFieldName
+
+          encodings.get(name) match {
+            case Some(geometryType) =>
+              builder.add(name, geometryType.clazz)
+            case None =>
+              val clazz = dataTypeToClass(field.getDataType)
+              builder.add(name, clazz)
+          }
+        }
+        // The default geometry is set as the first column name.
+        defaultGeometryColumn.foreach(builder.setDefaultGeometry)
+        typeName.foreach(builder.setName)
+
+        builder.buildFeatureType()
+    }
+  }
+
+  private val geometryTypeMap = scala.collection.Map(
```

Comment:
is this approach going to be case sensitive? do we care? (not sure how we handle that in normal sft parsing)

Reply:
Yes.  No, I don't think so.  Here's how we handle it: https://github.com/locationtech/geomesa/blob/main/geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/sft/SimpleFeatureSpec.scala#L274-L283
;)

Reply:
i guess we're more lenient with all the other types but not geoms... eh